creusot/tests/should_succeed/type_invariants/generated.rs

```diff
@@ -0,0 +1,8 @@
+extern crate creusot_contracts;
+use creusot_contracts::{invariant, *};
+
+pub struct List<T>(T, Option<Box<List<T>>>);
+
+pub fn use_list(l: List<i32>) {
+    proof_assert!(invariant::inv(l))
```

Comment:
I wonder if we couldn't write a more meaningful test? 

At the very least I think it would be a good idea to add an enum with struct fields, a struct with named fields and a type containing a lifetime.

Reply:
Yes, good idea.